File: 6508.py
from os import system
while(1):
    n=int(input())
    if n==0: break
    bowl=list(map(int,input().split()))
    count=0
    upOrDown=True #up: True down: False
    i=0
    while(sum(bowl)!=0):
        if i==0: bowl[i]-=1 if bowl[i]>0 else 0
        else:
            if bowl[i]>0:
                bowl[i]-=1
        count+=1
        print(*bowl)
        system("pause")
        i=i+1 if i+1<n else 0
        # i=i+1 if upOrDown else i-1
        # if i==n: i,upOrDown=i-2, False
        # elif i==-1: i,upOrDown=i+2, True
    print(count)

File: 6508.c
#include <stdio.h>
int main(){
    while(1){
        int n;
        scanf("%d", &n);
        if (n==0) break;
        int bowl[50];
        int i;
        for (i=0; i<n; i++) scanf("%d", &bowl[i]);
        long long count=0;
        i=0;
        int j;
        int sum=50000;
        while(sum!=0){
            if (bowl[i]==0) { 
                i=0; 
                continue; 
            }
            if (i==0){
                bowl[i]=bowl[i]>0 ? bowl-1 : 0;
                count+=1;
            }
            else {
                if (bowl[i]>0) {
                    bowl[i]-=1;
                    for (j=0; j<i; j++) bowl[j]+=1;
                    count++;
                }
            }
            i=i+1<n ? i+1 : 0;
            for (j=0; j<n; j++) sum+=bowl[j];
        }
        printf("%lld\n",count);
    }
    return 0;
}